### ISSUE TITLE
Proposal: @frogbotai/plugin-capture — opt-in request/response capture with per-key policy (blob storage, sampling, retention)

### ISSUE THREAD

**Issue by colby-trend** [frogbotai/frogbot#49]
## Motivation

Today the gateway stores usage metadata only (`usage_logs`: tokens, cost, model, key, user, request_id — no content). That's the right always-on default, but there's no supported way to capture request/response payloads at all, and several real consumers want them:

- **Debugging/support** — "why did this request fail / cost $4" needs the payload for *specific* requests, briefly. Capture-on-error plus opt-in capture with short retention covers it.
- **Security/compliance** — cuts both ways. Some orgs want a content audit trail for incident review; just as many make "prove you do *not* retain prompts" a procurement requirement. Zero-retention is a selling point, not a gap — which means capture must be a **policy switch, per API key, defaulting to off**, never always-on behavior.
- **Training data / evals** — for orgs that want to collect their own traffic as fine-tuning data, raw (request, response) pairs are exactly the SFT unit: the request body is the precise input the model saw (system prompt, tools, params included) and the response is the target.

This is also where the incumbents converged (LiteLLM, Helicone, Portkey): metadata always, content optional/sampled/TTL'd.

## Why a plugin, not core config

Matches the existing split: usage logging is core because it's universal; api-keys is a plugin because it's policy. Capture is pure policy. The gateway hook surface already carries everything a plugin needs:

- `beforeUpstream` exposes the **canonical, cross-provider request** — `messages`, `system`, `tools`, `params` — pre-translation, after other hooks have mutated it. Capturing here means one uniform dataset format regardless of upstream provider, and it records what was *actually sent* (post-mutation), which is what training fidelity requires.
- `afterUpstream` carries `response`, `usage`, `finishReason` (streaming currently omits `response` — see #48, which this proposal depends on).
- The mutable `context` bag is the sanctioned hook-to-hook channel (it's how auth hands the key to usage billing today).
- `usage_logs.request_id` is already indexed — a free join key between metadata and blobs. **No schema changes needed.**

## Design

**`@frogbotai/plugin-capture`** — injects two `ai.hooks` entries via config transform:

1. `beforeUpstream`: snapshot `{ messages, system, tools, params }` into the context bag. Language operations only (`chat.completions`, `messages`, `responses`) — modality routes don't expose inputs, and that's fine.
2. `afterUpstream` / `afterError`: evaluate policy, then fire-and-forget one blob write. Same failure posture as `logUsage`: log the error, never fail the request.

**Blob shape.** One compressed object per request, keyed by request id:

```
{requestId}.json.zst
{ requestId, model, provider, operation, params, messages, tools, system,
  response, finishReason, usage, error?, capturedAt }
```

**Storage adapter.** `storage: { put(key, bytes) }` plugin option — filesystem directory as the dev default, S3 (or any object store) for real deployments. Blobs never go in the OLTP database; the `usage_logs` row stays small and is the queryable index (model, key, user, cost, time) for filtering captures.

**Per-key policy.** When `plugin-api-keys` is present, add a `capture` field to the keys collection:

```
capture: off | errors-only | sample | full     (+ sampleRate when sample)
```

plus a global default in plugin options and a `context.capture === false` escape hatch mirroring the existing `trackUsage === false` convention. Default **off**. The admin field label should say plainly that requests through the key are recorded — this makes zero-retention and capture tiers of one visible switch, per key, which is the enterprise-shaped version (finance never needs content; compliance picks a side per org; debugging uses errors-only; training/eval teams opt keys in).

**Retention.** A periodic cleanup job with a retention-window option (e.g. 7–30 days for debugging-tier captures; `null` = keep, for training-tier), not per-record timers. Training captures are an asset; debugging captures are a liability past their window.

## Non-goals (deliberate)

- **No dedup / prefix-sharing at capture time.** Stateless clients (opencode etc.) re-send the whole conversation every turn, so raw capture is quadratic in stored bytes — but repeated JSON prefixes are zstd's best case, and dedup at write time means building fragile stitching heuristics on the hot path (clients compact/edit history; prefix-hash schemes fork). Capture dumb and faithful; dedup, thread-stitching, and dataset assembly are offline batch jobs that read the blobs and can be re-run when the logic improves.
- **No PII/secret scrubbing at capture.** Scrubbing is destructive and scrubbers have false positives; do it in the dataset pipeline where it's re-runnable. The plugin's job is policy + faithful bytes.
- **No analytics/reporting** — separate concern, filed separately.

## Dependencies

- #48 — streaming `afterUpstream` must carry the assembled `response`, or capture gets nothing for the dominant streaming path.
